shared/clarifai_shared/embedding/models.py

```diff
@@ -0,0 +1,317 @@
+"""
+Embedding module for ClarifAI utterance chunks.
+
+This module handles the generation of embeddings for utterance chunks using
+configurable models via LlamaIndex. Supports HuggingFace models including
+BERT-based sentence transformers.
+
+Key Features:
+- Configurable embedding models from clarifai.config.yaml
+- Batch processing for efficiency
+- Automatic device detection (CPU/GPU)
+- Integration with LlamaIndex embedding abstractions
+- Structured logging with service context
```

Comment:
unlike chunking.py, there is no documentation we could point people to to learn more? I feel like the pattern we are creating here is roughly similar (in size/complexity and extensibility), to the plugins pattern, which has its own documentation.

If you do create one, consider grouping them in docs/guides/ rather than leaving the plugin one as the only file in the docs/plugins/ folder.